File: src/java/magmac/app/compile/rule/split/LastSelector.java
package magmac.app.compile.rule.split;

import magmac.api.Option;
import magmac.api.Tuple2;
import magmac.api.collect.list.List;
import magmac.api.iter.collect.Joiner;

public class LastSelector implements Selector {
    @Override
    public Option<Tuple2<String, String>> select(List<String> list) {
        return list.popLast().map(tuple -> {
            String joined = tuple.left().iter().collect(new Joiner(" ")).orElse("");
            return new Tuple2<String, String>(joined, tuple.right());
        });
    }
}

File: src/web/magmac/app/compile/rule/split/LastSelector.ts
export class LastSelector {
	 LastSelector( delimiter : String) : public {this.delimiter=delimiter;;}
	public select( list : List<String>) : Option<Tuple2<String, String>> {return list.popLast( ).map( this.merge);;}
	private merge( tuple : Tuple2<List<String>, String>) : Tuple2<String, String> { let joined : var=tuple.left( ).iter( ).collect( new Joiner( this.delimiter)).orElse( "");return new Tuple2<String, String>( joined, tuple.right( ));;}
}
